src/sdk/pynni/nni/nas/pytorch/darts/mutator.py

```diff
@@ -3,16 +3,19 @@
 from torch.nn import functional as F
 
 from nni.nas.pytorch.mutables import LayerChoice
-from nni.nas.pytorch.mutator import PyTorchMutator
+from nni.nas.pytorch.mutator import Mutator
 
 
-class DartsMutator(PyTorchMutator):
+class DartsMutator(Mutator):
 
     def before_build(self, model):
         self.choices = nn.ParameterDict()
+        self.register_on_init_hook(LayerChoice, self.on_init_layer_choice)
+        self.register_on_forward_hook(LayerChoice, self.on_forward_layer_choice)
```

Comment:
I agree this is a better design, but probably we should include these two lines in `__init__`.